src/commands/mod.rs

```diff
@@ -792,14 +792,14 @@ struct RebaseArgs {
     /// Each specified revision will become a direct child of the destination
     /// revision(s), even if some of the source revisions are descendants
     /// of others.
-    #[arg(long, short)]
+    #[arg(long, short, default_value = "@-")]
     source: Vec<RevisionArg>,
     /// Rebase only this revision, rebasing descendants onto this revision's
     /// parent(s)
     #[arg(long, short)]
     revision: Option<RevisionArg>,
     /// The revision(s) to rebase onto (can be repeated)
-    #[arg(long, short, required = true)]
+    #[arg(long, short, default_value = "@-")]
```

Comment:
We usually default to `@`, not `@-` (e.g. `-b` defaults to `@` in `cmd_rebase`), but I admit it is a bit weird to use `@` in this case. I think this suggests that our current `@` vs `@-` distinction is a UX wart, and we should find a way to fix it at the root.

Until then, I'd prefer to use `@` for consistency nevertheless.

Reply:
For people used to git, it'd be strange to have a default here without having a concept of upstream branches. In that spirit, the best we could do now would be to default to `main|master`.

Reply:
Should we add a configurable default rebase destination like git-branchless has?

Reply:
(Maybe you're saying the same thing) I'd be interested in having a trunk-based mode. git-branchless does that, which allows a lot of wonderful simplifying assumptions to be made (e.g. where branches should be rebased to by default, which remote branches you care about, log outputting). We definitely shouldn't make it _required_, but it would be a lot easier for most folks I'd think.

Reply:
> Should we add a configurable default rebase destination like git-branchless has?

To be clear, the destination for [`git move`](https://github.com/arxanas/git-branchless/wiki/Command:-git-move) (the rebase equivalent) defaults to `@`, not the main branch. The different command [`git sync`](https://github.com/arxanas/git-branchless/wiki/Command:-git-sync) is used to rebase on top of the main branch, but the default source is all `draft()` commits.

Rebasing on top of `@-` is meant to simulate the git-branchless workflow of grabbing another commit stack from somewhere else and integrating it into the current commit stack. I suppose rebasing on top of `@` would also work, although I don't know if I usually want to do that.

> For people used to git, it'd be strange to have a default here without having a concept of upstream branches. In that spirit, the best we could do now would be to default to main|master.

@ilyagr I don't understand this; why would we need a concept of upstream branches (or branches at all) to have a default for `rebase`, and why would we want to default to `main`/`master` in that case?

Reply:
If I remember correctly, in git a plain `git rebase` with no arguments is loosely equivalent to `jj rebase -b current_branch -d current_branch@upstream`.  This makes `git pull --ff-only && git rebase` equivalent to `git pull --rebase`.

I might be wrong -- not having to remember these things is one of the reasons is one of the reasons I am excited about jj.

It seemed me that `jj rebase -b @ -d main|master` is a loose approximation to the above behavior that jj can currently simulate. jj is limited in its ability to simulate that since it doesn't have a notion of a "current branch", let alone "upstream for the current branch".